### DISCUSSION TITLE
Rendering templates documentation edit

### DISCUSSION THREAD

**Discussion by jaketanis** [pallets/flask#5484] [Ideas]
I am working through the quickstart guide and testing out templates, and this code snippet confused me a bit:

```py
from flask import render_template

@app.route('/hello/')
@app.route('/hello/<name>')
def hello(name=None):
    return render_template('hello.html', name=name)
```
I wasn't sure why `name=name` was necessary. I learned that the first `name` is referring to the variable in the referenced html file:
```html
<!doctype html>
<title>Hello from Flask</title>
{% if name %}
  <h1>Hello {{ name }}!</h1>
{% else %}
  <h1>Hello, World!</h1>
{% endif %}
```
This may be trivial, but since it confused me, I thought it might of confused others. I wanted to suggest something like this: 

```py
@app.route("/hello/")
@app.route('/hello/<name>')
def hello(name=None):
    return render_template('hello.html', person=name)
```
with the html file looking like this:
```html 
<!doctype html>
<title>Hello from Flask</title>
{% if person %}
  <h1>Hello {{ person }}!</h1>
{% else %}
  <h1>Hello, World!</h1>
{% endif %}
```
I think this helps differentiate the variable in the html file from the url variable. Let me know your thoughts. 😄

**Comment by davidism** [comment]
Makes sense to me, did you want to make a PR?

**Reply by jaketanis** [reply]
> Makes sense to me, did you want to make a PR?

Yes, just wanted to bring it up here first because the contribution guidelines said to. I'll go ahead and do that!
